src/backtrace.rs

```diff
@@ -0,0 +1,48 @@
+use std::{collections::HashSet, path::Path};
+
+use object::read::File as ElfFile;
+use probe_rs::{config::RamRegion, Core};
+
+use crate::{Outcome, VectorTable};
+
+mod pp;
+mod symbolicate;
+mod unwind;
+
+/// (virtually) unwinds the target's program and prints its backtrace
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn print(
+    core: &mut Core,
+    debug_frame: &[u8],
+    elf: &ElfFile,
+    vector_table: &VectorTable,
+    sp_ram_region: &Option<RamRegion>,
+    live_functions: &HashSet<&str>,
+    current_dir: &Path,
+    force_backtrace: bool,
+    max_backtrace_len: u32,
```

Comment:
I think I agree with Clippy here, it would be nice to group some of these arguments in a struct (maybe have one for backtrace options, one for stuff extracted from the ELF)

Reply:
I agree. For now I have grouped the backtrace settings into a struct. That removes 2 arguments and barely passes the clippy lint.

I'll leave the refactor of grouping ELF stuff into its own struct as follow-up work. In my mind, there should a (fallible?) operation that goes from `object::File` into `our::ProcessedElf` which is then passed by reference everywhere. That op should be done early in the program to detect issues like missing debug info early on.